module_utils/network_lsr/argument_validator.py

```diff
@@ -2012,3 +2083,101 @@ def _ipv6_is_not_configured(connection):
                             "match.path is not supported by the running version of "
                             "NetworkManger.",
                         )
+        self.validate_route_tables(connection, idx)
+
+
+class IPRouteUtils(object):
+
+    ROUTE_TABLE_ALIAS_RE = re.compile("^[a-zA-Z0-9_.-]+$")
+
+    @classmethod
+    def _parse_route_tables_mapping(cls, file_content, mapping):
+        for line in file_content.split(b"\n"):
+
+            #
+            # We are a bit more forgiving, but the aliases must be nice UTF-8 names.
+
+            # iproute2 only skips over leading ' ' and '\t'. We are more forgiving,
+            # and strip all leading+trailing whitespace.
+            line = line.strip()
+
+            # skip empty lines or comments
+            if not line:
+                continue
+            if line[0] == ord(b"#"):
+                continue
+
+            # Split at the first space. We accept here any whitespace (unlike iproute2).
+            # Also, this will strip any leading/trailing whitespaces (unlike iproute2)
+            ll = line.split(maxsplit=1)
+            if len(ll) != 2:
+                continue
+            line1 = ll[0]
+            line2 = ll[1]
+
+            i = line2.find(b" ")
+            if i == -1:
+                i = line2.find(b"\t")
+            if i >= 0:
+                # again, iproute2 only accepts one ' ', followed by '#'. We accept any number of
+                # space and any kind of whitespace (the split() above already partly took care
+                # of those).
+                line3 = line2[i + 1 :].strip()
+                if line3 and line3[0] != ord(b"#"):
+                    continue
+                line2 = line2[:i]
+
+            # unlike iproute2, convert to UTF-8 and only accept benign characters.
+            try:
+                line2 = line2.decode("utf-8")
+            except Exception:
+                continue
+
+            if not cls.ROUTE_TABLE_ALIAS_RE.match(line2):
+                continue
+
+            i = None
+            try:
+                i = int(line1)
+            except Exception:
+                if line.startswith(b"0x"):
+                    try:
+                        i = int(line1[2:], 16)
+                    except Exception:
+                        pass
+            if i is None or i < 0 or i > 0xFFFFFFFF:
+                continue
+
+            mapping[line2] = i
+
+    @classmethod
+    def _parse_route_tables_mapping_from_file(cls, filename, mapping):
+        try:
+            with open(filename, "rb") as f:
+                file_content = f.read()
+        except Exception:
+            return
+        cls._parse_route_tables_mapping(file_content, mapping)
+
+    @classmethod
+    def get_route_tables_mapping(cls):
```

Comment:
It is not clear to me why is this using the `@classmethod` decorator. It is a problem if you create an instance? I think it would simplify the code.

Reply:
Notice that the method `get_route_tables_mapping` accesses other methods in the class,  so it is better to use classmethod here. And the benefits of using the `@classmethod` is to call the method without instantiate the class.